### PR TITLE
Add progress bar to skill badges

### LINKED ISSUE #152: Frontend show real skill levels

### PR DESCRIPTION
**Pull request by mardom1**
# Description of changes

image

Closes #152 

## How has this been tested?

Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.

## Checklist before requesting a review

- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: app/courses/[courseId]/student.tsx [hunk 1]
```diff
@@ -358,29 +351,30 @@ function StudentSection({
 
   // Workaround until the backend calculates the progress
```
Comment (SkyDiverStar) [inline app/courses/[courseId]/student.tsx:352]: Is it still a workaround? If not, we should remove this comment

**Comment by SkyDiverStar** [comment]
grafik
Ich habe beide Sections erledigt, laut dem Bild, aber nur die Hälfte des Chapters. Wie kommt das? Haben wir einen Fehler bei der Berechnung gemacht oder kommt dass durch das Backend?
Ist ein Bild aus dem Kurs PSE

**Comment by SkyDiverStar** [comment]
Habe mehrmals diverse Flashcards und Quizzes gemacht (da die Medias, bei mir nicht laden aus irgendeinem Grund). Die Skilllevels werden nicht aktualisiert oder besser gesagt ich steige kein Level auf. Woran könnte das liegen?

**Comment by mardom1** [comment]
Der Course Progress wird derzeit noch im Frontend berechnet (offensichtlich fehlerhaft). Der Plan ist aber das ins Backend zu verschieben. Lohnt sich also nicht das jetzt zu fixen.

**Comment by mardom1** [comment]
Das mit den Skill Levels musst du die Backend Leute fragen.

File: app/courses/[courseId]/student.tsx (hunk 1 above)
Comment (mardom1) [reply app/courses/[courseId]/student.tsx:352]: Immer noch ein Workaround. Nur ein etwas schönerer

Comment (SkyDiverStar) [reply app/courses/[courseId]/student.tsx:352]: Ok verstanden
